cli/flox/src/commands/install.rs

```diff
@@ -226,4 +238,168 @@ impl Install {
             err => apply_doc_link_for_unsupported_packages(err).into(),
         }
     }
+
+    /// Generate warnings to print to the user about unfree and broken packages.
+    fn generate_warnings(
+        locked_packages: &[LockedPackageCatalog],
+        packages_to_install: &[PackageToInstall],
+    ) -> Vec<String> {
+        let mut warnings = vec![];
+
+        // There could be multiple packages with the same install_id but different systems.
+        // A package could be broken on one system but not another.
+        // So just keep track of which install_ids we've warned for.
+        // TODO: does the warning need to take system into account?
+        let mut warned_unfree = HashSet::new();
+        let mut warned_broken = HashSet::new();
+        for locked_package in locked_packages.iter() {
+            // If unfree && just installed && we haven't already warned for this install_id,
+            // warn that this package is unfree
+            if locked_package.unfree == Some(true)
+                && packages_to_install
+                    .iter()
+                    .any(|p| locked_package.install_id == p.id)
+                && !warned_unfree.contains(&locked_package.install_id)
+            {
+                warnings.push(format!("The package '{}' has an unfree license, please verify the licensing terms of use", locked_package.install_id));
+                warned_unfree.insert(&locked_package.install_id);
+            }
+
+            // If broken && just installed && we haven't already warned for this install_id,
+            // warn that this package is broken
+            if locked_package.broken == Some(true)
+                && packages_to_install
+                    .iter()
+                    .any(|p| locked_package.install_id == p.id)
+                && !warned_broken.contains(&locked_package.install_id)
+            {
+                warnings.push(format!("The package '{}' is marked as broken, it may not behave as expected during runtime.", locked_package.install_id));
+                warned_broken.insert(&locked_package.install_id);
+            }
+        }
+        warnings
+    }
```

Comment:
wrt to my comment above i think we could implement this checking once, rather than twice, once for raising errors and once for printing warnings.

Reply:
I'm not quite sure how we would simplify that? Since they're checking different things